### PR TITLE
Fixes #21629: std.csv report one record on empty input

### PR DESCRIPTION
**Pull request by burner**
https://issues.dlang.org/show_bug.cgi?id=21629

### CODE REVIEW

File: std/csv.d [hunk 1]
```diff
@@ -1750,3 +1769,19 @@ if (isSomeChar!Separator && isInputRange!Range
...
+    struct Reccord
+    {
+        string a;
+        string b;
+    }
+
+    string input = "";
+    foreach (it; csvReader!Reccord(input))
+    {
+        assert(false, "We shouldn't make it here");
```
Comment (deadalnix) [inline std/csv.d:1785]: Maybe you could count loop iterration and assert the count is zero so that the linter can be happy.

Comment (burner) [reply std/csv.d]: done

**Comment by deadalnix** [comment]
LGTM. I don't know what the next step is but as far as I'm concerned, this is good to go.

Maybe adding a test case with a CSV header then no data after and and checking that this also doesn't return any row, but this also could be done in another PR if that is hard, there is no reason to stop this due to scope creep.

**Comment by burner** [comment]
> Maybe adding a test case with a CSV header then no data after and and checking that this also doesn't return any row, but this also could be done in another PR if that is hard, there is no reason to stop this due to scope creep.

That is actually tested.

**Comment by deadalnix** [comment]
The current test checks csv("", ["a", "b"]) and verify it is empty

But afaik, this is not tested:
csv("a,b\n", ["a", "b"])

This also should return an empty range.

**Comment by burner** [comment]
ah, I miss read. Looking at the other tests this should be handled already.

**Review by RazvanN7 — APPROVED** [review]

**Comment by burner** [comment]
thank you @RazvanN7 @Geod24 @deadalnix

**Pull request merged**